File: ruby-gem/lib/models/social_media.rb
# frozen_string_literal: true

# Class from DFC Business Ontology: #SocialMedia
require_relative '../core/semantic_object'
require_relative 'what_subject'


module DfcLinkmlConnector
  module Models
    class SocialMedia < WhatSubject
      SEMANTIC_TYPE = "dfc-b:SocialMedia".freeze

      # @return [String]
      attr_accessor :website_page

      # @return [String]
      attr_accessor :social_media_of

      # @param semanticId [String]
      # @param date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, websitePage: nil, socialMediaOf: nil
      def initialize(semanticId, date: nil, description: nil, name: nil, characteristicOf: nil, dimension: nil, websitePage: nil, socialMediaOf: nil)
        super(semanticId, date: date, description: description, name: name, characteristicOf: characteristicOf, dimension: dimension)
        @website_page = websitePage
        @social_media_of = socialMediaOf
        self.semanticType = "dfc-b:SocialMedia"
        registerSemanticProperty("dfc-b:websitePage", &method("website_page")).valueSetter = method("website_page=")
        registerSemanticProperty("dfc-b:socialMediaOf", &method("social_media_of")).valueSetter = method("social_media_of=")
      end

      Core::SemanticObject.type_registry[SEMANTIC_TYPE] = self
    end
  end
end

File: typescript-connector/src/models/SocialMedia.ts
// Class from DFC Business Ontology: #SocialMedia
import { SemanticObject } from "../core/SemanticObject.js";
import { WhatSubject, type WhatSubjectParams } from "./WhatSubject.js";

export interface SocialMediaParams extends WhatSubjectParams {
  websitePage?: string;
  socialMediaOf?: string;
}
export class SocialMedia extends WhatSubject {
  static get SEMANTIC_TYPE(): string {
    return "dfc-b:SocialMedia";
  }

  websitePage?: string;
  socialMediaOf?: string;

  constructor(
    semanticId: string,
    params?: SocialMediaParams,
  ) {
    super(semanticId, params);
    this.websitePage = params?.websitePage;
    this.socialMediaOf = params?.socialMediaOf;
    this.semanticType = SocialMedia.SEMANTIC_TYPE;
    this.registerSemanticProperty("dfc-b:SocialMedia:website_page", () => this.websitePage);
    this.registerSemanticProperty("dfc-b:SocialMedia:social_media_of", () => this.socialMediaOf);
  }
  static {
    SemanticObject.typeRegistry.set(SocialMedia.SEMANTIC_TYPE, SocialMedia);
  }
}
